### ISSUE TITLE
NetworkInterface.GetAllNetworkInterfaces() on Ubuntu 14.04 unhandled exception FileNotFoundException

### ISSUE THREAD

**Issue by allevyMS** [dotnet/corefx#9761]
I get the following exception when calling GetAllNetworkInterfaces on Ubuntu:

```
Unhandled Exception: System.IO.FileNotFoundException: Could not find file '/sys/class/net/em1:avahi/operstate'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Interop.CheckIo[TSafeHandle](TSafeHandle handle, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.UnixFileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
   at System.IO.UnixFileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.File.InternalReadAllText(String path, Encoding encoding)
   at System.Net.NetworkInformation.LinuxNetworkInterface.GetOperationalStatus(String name)
   at System.Net.NetworkInformation.LinuxNetworkInterface..ctor(String name)
   at System.Net.NetworkInformation.LinuxNetworkInterface.GetOrCreate(Dictionary`2 interfaces, String name)
   at System.Net.NetworkInformation.LinuxNetworkInterface.<>c__DisplayClass5_0.<GetLinuxNetworkInterfaces>b__0(String name, IpAddressInfo* ipAddr, IpAddressInfo* maskAddr)
   at Interop.Sys.EnumerateInterfaceAddresses(IPv4AddressDiscoveredCallback ipv4Found, IPv6AddressDiscoveredCallback ipv6Found, LinkLayerAddressDiscoveredCallback linkLayerFound)
   at System.Net.NetworkInformation.LinuxNetworkInterface.GetLinuxNetworkInterfaces()
   at System.Net.NetworkInformation.NetworkInterface.GetAllNetworkInterfaces()
```

I’m using package: "System.Net.NetworkInformation": "4.1.0-rc3-24210-09"

This isn’t reproducible on all Ubuntu machines, since on some it works while on others it throws the above exception

Thanks,

Alon

**Comment by allevyMS** [comment]
An additional observation: the machine on which the call fails has the avahi-daemon running while the one on which it works doesn't.

**Comment by davidsh** [comment]
cc: @stephentoub

**Comment by stephentoub** [comment]
cc: @mellinoe

**Comment by mellinoe** [comment]
This should have been fixed by #9778. If this still repros with a new version of the package, please re-open.
